ui/pages/keychains/restore-vault.js

```diff
@@ -36,9 +43,21 @@ class RestoreVaultPage extends Component {
       createNewVaultAndRestore,
       leaveImportSeedScreenState,
       history,
+      // eslint-disable-next-line no-shadow
+      isSocialLoginFlow,
     } = this.props;
 
     leaveImportSeedScreenState();
+
+    if (isSocialLoginFlow) {
+      // reset oauth and onboarding state
+      await this.props.resetOAuthLoginState();
```

Comment:
we should reset resetOAuthLoginState  after successfully `createNewVaultAndRestore`.

Reply:
also the handleImport function lacks any error handling in general.